File: autowiring/src/main/kotlin/com/mthaler/bankapp/service/AccountStatementServiceImpl.java
package com.mthaler.bankapp.service;

import com.mthaler.bankapp.dao.AccountStatementDao;
import com.mthaler.bankapp.domain.AccountStatement;

import java.util.Date;

public class AccountStatementServiceImpl implements AccountStatementService {
	private AccountStatementDao accountStatementDao;
	
	public void setAccountStatementDao(AccountStatementDao accountStatementDao) {
		this.accountStatementDao = accountStatementDao;
	}

	@Override
	public AccountStatement getAccountStatement(Date from, Date to) {
		return accountStatementDao.getAccountStatement(from, to);
	}
}

File: autowiring/src/main/kotlin/com/mthaler/bankapp/service/AccountStatementServiceImpl.kt
package com.mthaler.bankapp.service

import com.mthaler.bankapp.dao.AccountStatementDao
import com.mthaler.bankapp.domain.AccountStatement
import java.util.*

class AccountStatementServiceImpl : AccountStatementService {

    private var accountStatementDao: AccountStatementDao? = null

    fun setAccountStatementDao(accountStatementDao: AccountStatementDao?) {
        this.accountStatementDao = accountStatementDao
    }

    override fun getAccountStatement(from: Date?, to: Date?): AccountStatement? {
        return accountStatementDao!!.getAccountStatement(from, to)
    }
}